snapcraft/internal/remote_build/_launchpad.py

```diff
@@ -74,39 +119,80 @@ def login(self) -> None:
             version="devel",
         )
 
-    def create_snap(self, repository: str, archs: List[str]) -> None:
-        """Create a snap recipe."""
-        logger.debug("Create snap for {}".format(self._id))
-        # TODO: remove this after launchpad infrastructure is ready (LP #1827679)
-        url = repository.replace("git+ssh://", "https://")
-        snap = {
-            "name": self._id,
-            "owner": "/~" + self._user,
-            "git_repository_url": url,
-            "git_path": "master",
-            "auto_build": False,
-            "auto_build_archive": "/ubuntu/+archive/primary",
-            "auto_build_pocket": "Updates",
-        }
-
-        if archs:
-            snap["processors"] = ["/+processors/" + arch for arch in archs]
-
-        self._lp.snaps.new(**snap)
+    def get_git_repo_path(self) -> str:
+        return f"~{self._lp_user}/+git/{self._lp_name}"
+
+    def get_git_https_url(self, token: Optional[str] = None) -> str:
+        if token:
+            return f"https://{self._lp_user}:{token}@git.launchpad.net/~{self._lp_user}/+git/{self._lp_name}/"
+        else:
+            return f"https://{self._lp_user}@git.launchpad.net/~{self._lp_user}/+git/{self._lp_name}/"
```

Comment:
do we have a use case for tokenless? If not, remove

Reply:
Yes, `create_snap` uses it for the git_rpeository_url field.